### DISCUSSION TITLE
Plausible instance is completely down after update

### DISCUSSION THREAD

**Discussion by josefpolach** [plausible/analytics#3695] [Self-Hosted Support]
Hello, community,

I regret it a lot but I upgraded my plausible instance and it seems all is **lost**. :-( 

I pulled the last version using `git pull origin` and ran the docker-compose and it seems that everything is out of order now.

the main plausible container logs 
```
18:00:00.322 [error] Postgrex.Protocol (#PID<0.185.0>) failed to connect: ** (DBConnection.ConnectionError) tcp connect (plausible_db:5432): non-existing domain - :nxdomain` even though my database is set up correctly in docker-compose.yml.
```
and then
```
2024-01-16 18:07:10.087 UTC [1] FATAL:  database files are incompatible with server
2024-01-16 18:07:10.087 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 12, which is not compatible with this version 14.10.
```

the plausible_events_db writes networking problem which I am afraid I cant resolve I do not have a IPv6 support on the VM. 
```
2024.01.16 17:59:19.391939 [ 1 ] {} <Warning> Application: Listen [::]:9009 failed: Poco::Exception. Code: 1000, e.code() = 0, DNS error: EAI: Address family for hostname not supported (version 23.3.7.5 (official build)). If it is an IPv6 or IPv4 address and your host has disabled IPv6 or IPv4, then consider to specify not disabled IPv4 or IPv6 address to listen in <listen_host> element of configuration file. Example for disabled IPv6: <listen_host>0.0.0.0</listen_host> . Example for disabled IPv4: <listen_host>::</listen_host>
```

It seems that the easies way is to just run a new VM and run the Plausible again. But then I will loose all the data. Does anyone think there is a slight change to get to the data for past 2 years? I guess I just accept if not.

I am running plausible according to the documentation from the docker-compose.yml and using apache as a proxy.

Thanks to anyone for any kind of help 🙏

**Comment by ruslandoga** [comment]
👋 @josefpolach 

Worry not! I think these problems are rather easily resolvable. You can ignore the `plausible_events_db` warning. All we need to fix is the PostgreSQL error, which is also the result of `nxdomain` Plausible error (PostgreSQL didn't start, Docker didn't add it to its internal DNS resolver, Plausible app couldn't resolve it).

There are multiple ways to fix the PostgreSQL error. The first one is to switch back to `postgres:12`

In your [docker-compose.yml](https://github.com/plausible/hosting/blob/master/docker-compose.yml) you can change the image to the one that was there before and re-<kbd>docker compose up</kbd> the containers
```diff
  plausible_db:
    # supported versions are 12, 13, and 14
-   image: postgres:14-alpine
+   image: postgres:12-alpine
    restart: always
    volumes:
      - db-data:/var/lib/postgresql/data
    environment:
      - POSTGRES_PASSWORD=postgres
```

**Reply by ruslandoga** [reply]
Another way that dumps the data in an older PostgreSQL version and then restores it into a newer PostgreSQL version is described in https://github.com/plausible/hosting/blob/master/upgrade/postgres.md

**Reply by ruslandoga** [reply]
And then finally there is the `pg_upgrade` approach, which I've never used :) It requires to have binaries available for both the new and the old PostgreSQL versions so you can use one of these container images: https://github.com/tianon/docker-postgres-upgrade -- and follow their guides.

**Reply by josefpolach** [reply]
Thanks to your wisdom is all up and running! Thank you a lot! I went with changing the postgres from 14 to 12. 

But now it seems all the data from the past are gone. Can I kindly ask for your valuable advise one more time @ruslandoga ? Can the previous data be restored?

**Reply by ruslandoga** [reply]
@josefpolach sure!

You probably need to perform a data migration for the new tables. Please see https://github.com/plausible/analytics/discussions/3132

**Reply by josefpolach** [reply]
I followed the data migration as I did for the first time, before I changed the image version. But now without any errors. It's done then. Thanks a lot @ruslandoga
